include/boost/iterator/function_input_iterator.hpp

```diff
@@ -60,7 +56,7 @@ namespace iterators {
         public:
             function_object_input_iterator() {}
             function_object_input_iterator(Function & f_, Input state_ = Input())
-                : f(boost::addressof(f_)), state(state_) {}
+                : f(std::addressof(f_)), state(state_) {}
```

Comment:
I think, `boost::addressof` is still preferable as it doesn't require `<memory>`. Here and everywhere else.